### ISSUE TITLE
Setting a background ROI in ReflectometryReductionOneAuto with polarization corrections causes an error

### ISSUE THREAD

**Issue by rbauststfc** [mantidproject/mantid#35855]
**Original reporter:** Becky at ISIS

**Describe the bug**

When we perform a reduction using `ReflectometryReductionOneAuto` that requires polarization correction the algorithm follows these steps:

1) Perform the reduction as normal, which includes a step to sum the workspace to produce `IvsLam`. This step results in a workspace with a single spectrum.
1) Apply the polarization correction.
1) Repeat the reduction to recalculate `IvsQ`. This means that some (although not all) of the steps for producing `IvsLam` are repeated, as follows:
       - If we're summing in lambda - we first try to extract the spectrum numbers for the region of interest (signal and background ROIs) from the input workspace. By this point there is just a single spectrum in the input workspace. The `ProcessingInstructions` for the signal ROI will have been updated in method `ReflectometryReductionOneAuto3::createAlgorithmForGroupMember` to correctly account for this. However if `BackgroundProcessingInstructions` have been set then these are not updated, and it results in an error when they can't be found in the input workspace.
       - If we're summing in Q - we attempt the background subtraction using the original `BackgroundProcessingInstructions` and this causes an error because there is only the single spectrum in the input workspace.

Step 3 will complete successfully if you happen to have the `BackgroundProcessingInstructions` set to the spectrum number that remains in the summed workspace after step 1, although I don't know whether that would be correct scientifically.

**To Reproduce**

1) Open the ISIS Reflectometry GUI
1) On the Runs tab, set the instrument to `POLREF` and enter run `37125` and angle `2.3` into the first row of the processing table on the right.
1) On the Experiment Settings tab, enter `50-70` in the Background column of the Settings Lookup table.
1) Tick the Subtract Background checkbox.
1) Select either `Workspace` or `FilePath` from the Polarization Corrections drop-down and select the workspace or file path to the Polarization Efficiencies (ask Rachel if you don't have anything to test this with).
1) Back on the Runs tab, click to process the reduction. You should see the following error:

> ReflectometryISISLoadAndProcess started
> Sample Log "HEATER" and 13 others contain invalid values, click "Show Sample Logs" for details.
> Log "running" has 30 entries removed due to duplicated time. 
> ReflectometryReductionOneAuto started
> ReflectometryReductionOneAuto successful, Duration 0.50 seconds
> ReflectometryReductionOneAuto started
> ReflectometryReductionOneAuto successful, Duration 0.53 seconds
> GroupWorkspaces started
> GroupWorkspaces successful, Duration 0.00 seconds
> GroupWorkspaces started
> GroupWorkspaces successful, Duration 0.00 seconds
> GroupWorkspaces started
> GroupWorkspaces successful, Duration 0.00 seconds
> PolarizationEfficiencyCor started
> PolarizationEfficiencyCor successful, Duration 0.11 seconds
> ReflectometryReductionOneAuto started
> Error in execution of algorithm ReflectometryReductionOne:
> Could not find spectrum number in any spectrum.
> Error in execution of algorithm ReflectometryReductionOneAuto:
> Could not find spectrum number in any spectrum.
> Error in execution of algorithm ReflectometryISISLoadAndProcess:
> Could not find spectrum number in any spectrum.
>   at line 109 in 'C:/repos/mantid/Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/ReflectometryISISLoadAndProcess.py'
>   caused by line 642 in 'C:/repos/mantid/Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/ReflectometryISISLoadAndProcess.py'
> Error starting batch algorithm
> Got error from algorithm "ReflectometryISISLoadAndProcess"

**Expected behavior**

The reduction should complete successfully having applied both the background subtraction and polarization correction correctly. We will need to confirm what the correct workflow should be when using polarization corrections with background subtraction (and take the opportunity to check the further processing that is happening after the polarization correction step) to know exactly what needs to be changed.

**Comment by rbauststfc** [comment]
Information so far from Becky, but this is still to be confirmed:

I had thought that the second run of RROA was purely to produce the conversion to Q of the corrected lambda workspace and not do any of the other steps but sounds like it has been trying to do more than that. The background subtraction etc. will have been carried out in the first run of RROA to produce the Lam workspace so we don't want it to be done again.

I think all we need is the part to go from Lam to Q (unless I've missed something else that needs to be done).

**Comment by rbauststfc** [comment]
After reviewing the reduction steps with Becky, the following should be skipped on the run through that happens after the polarization corrections have been applied:

- Skip the sum banks step in `ReflectometryReductionOneAuto` on the second run.
- Skip the background subtraction step in `ReflectometryReductionOne`.

While we're making changes, we should also change the first step of `ReflectometryReductionOne::makeIvsLam` that extracts the region of interest from the workspace when summing in lambda so that it is conditional on the background subtraction checkbox rather than processing instructions, as it should only be required if background subtraction is being performed.

These changes will provide a quick fix, but longer term, the existing implementation seems to have a lot of redundant processing and there is a high risk of unintentionally adding steps that shouldn't be performed after the polarization correction. A separate issue will be opened to look into refactoring this.
